pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/JdbiMailAddress.java

```diff
@@ -94,4 +111,60 @@ default int updateAddress(String guid, MailAddress addressWithUpdatedFields,
     @UseStringTemplateSqlLocator
     boolean deleteDefaultAddressByParticipantId(@Bind("participantId") Long participantId);
 
+
+    /**
+     * Reducer for reading non-default MailAddresses of a specified set of Participants
+     */
+    class BulkFindNonDefaultMailAddressesReducer implements LinkedHashMapRowReducer<String, NonDefaultMailAddressesWrapper> {
+
+        @Override
+        public void accumulate(Map<String, NonDefaultMailAddressesWrapper> container, RowView view) {
+            try {
+                String participantGuid = view.getColumn("participant_guid", String.class);
+                long instanceId = view.getColumn("instance_id", Long.class);
+                MailAddress mailAddress = new MailAddress(
+                        view.getColumn("address_id", Long.class),
+                        view.getColumn("address_guid", String.class),
+                        view.getColumn("name", String.class),
+                        view.getColumn("street1", String.class),
+                        view.getColumn("street2", String.class),
+                        view.getColumn("city", String.class),
+                        view.getColumn("state", String.class),
+                        view.getColumn("country", String.class),
+                        view.getColumn("zip", String.class),
+                        view.getColumn("phone", String.class),
+                        view.getColumn("pluscode", String.class),
+                        view.getColumn("description", String.class),
+                        DsmAddressValidationStatus.getByCode(view.getColumn("validationStatus", Integer.class)),
+                        false
+                );
+                container.computeIfAbsent(participantGuid, NonDefaultMailAddressesWrapper::new).unwrap().put(instanceId, mailAddress);
+            } catch (Exception e) {
+                throw new DDPException("Error during parsing a DB result with non-default MailAddresses ", e);
```

Comment:
I recommend throwing `DaoException` when we're in DAO code.

Reply:
Fixed